### PR TITLE
Abdulhakim/require source in payload

### PR DESCRIPTION
**Pull request by hakimms**
[Bug 2161566](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2161566): Require Source field in Payload

### CODE REVIEW

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,9 @@
 
 ### Bugs Fixed
 
-- Updated ODataType signature - 2433332
-- Empty or null response actions will throw a bad response - 2161553
+- Updated ODataType signature - https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2433332
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md:7]: This ChangeLog is meant to be displayed to the general public. The links aren't helpful to them. Let's remove

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/Framework/AuthenticationEventMetadataTests.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        [TestCaseSource(nameof(TestScenarios))]
+        public void TestRequestCreateInstance(object testObject, string message, bool success)
+        {
+            string payload = testObject.ToString();
+            AuthenticationEventMetadata eventMetadata = AuthenticationEventMetadataLoader.GetEventMetadata(payload);
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, $"http://localhost:7278/runtime/webhooks/customauthenticationextension?code=opVsbQul8-MsRuM9x6yVoghE2Xda5G-MeV_Ybv1MdEfuAzFuVtpEpg==&functionName=onTokenIssuanceStart");
+
+            if (success == false)
+            {
+                var ex = Assert.Throws<ValidationException>(() => eventMetadata.CreateEventRequestValidate(requestMessage, payload, string.Empty));
```
Comment (HarmanDhunna) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/Framework/AuthenticationEventMetadataTests.cs:26]: could you also Assert on the exception message to make sure it's what we expect.

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md (hunk 1 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/CHANGELOG.md]: Removed

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/Framework/AuthenticationEventMetadataTests.cs (hunk 2 above)
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/tests/Framework/AuthenticationEventMetadataTests.cs:26]: Done, updated the other test cases as well

**Review by HarmanDhunna — APPROVED** [review]

**Pull request merged**